src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,325 @@
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+
+    public static final int MAX_CAPACITY = 100;
+    public static final int INDEX_PAST_RECORD_STATUS = 4;
+    public static final String pastRecordPath = "./iP/iPoutput.txt";//file path from root file
+
+
+    public static void main (String[] args) throws IOException {
+        printWelcomeMessage ();
+        decodingUserCommands ();
+    }
+
+    private static void printLine () {
+        System.out.println ("----------------------------------"
+                + "-------------------------------------------");
+    }
+
+    private static void printBye () {
+        System.out.println ("Bye. Hope to see you again soon!");
+    }
+
+    private static void printGreeting () {
+        String greeting = "Hello! I'm Duke\n"
+                + "What can I do for you?\n";
+        System.out.print (greeting);
+        printLine ();
+    }
+
+    /**
+     * Reads user input and adds it to the task list accordingly
+     * Based on the input, new task of different types (e.g todo, event, deadline)
+     * will be created accordingly
+     */
+    private static void decodingUserCommands () throws IOException {
+        String input;
+        int listSize = 0;
+        ArrayList<String> list = new ArrayList<String> ();
+        Task[] tasks = new Task[MAX_CAPACITY];
+
+        //importing pastRecord before reading user commands
+        printImportingMessage ();
+        File pastRecord = new File (pastRecordPath);
+        try {
+            Scanner scanner = new Scanner (pastRecord);
+            while (scanner.hasNextLine ()) {
+                listSize = importPastRecord (listSize, list, tasks, scanner);
+            }
+            printDoneLoadingMessage ();
+        } catch (FileNotFoundException e) {
+            System.out.println ("There is no past record! You are a 1st time user");
+            printLine ();
+        }
+
+        //Reading user commands after importing pastRecord
+        Scanner in = new Scanner (System.in);
+        input = in.nextLine ();
+        while (! input.equals ("bye")) {
+            if (input.equals ("list")) {
+                printList (list, tasks);
+            } else if (input.startsWith ("done")) {
+                setDone (input, list, tasks);
+            } else if (input.startsWith ("todo")) {
+                listSize = setTodo (input, listSize, list, tasks);
+            } else if (input.startsWith ("event")) {
+                listSize = setEvent (input, listSize, list, tasks);
+            } else if (input.startsWith ("deadline")) {
+                listSize = setDeadline (input, listSize, list, tasks);
+            } else if (input.startsWith ("delete")) {
+                listSize = setDelete (input, listSize, list, tasks);
+            } else {
+                System.out.println ("\u2639 " + "OOPS!!! I'm sorry, but I don't know what that means :-(");
```

Comment:
magic string u2639? please refert to https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well